### ISSUE TITLE
Is delegating to enum in the target range

### ISSUE THREAD

**Issue by gfreezy** [Kobzol/rust-delegate#44]
Proposed usage is like below:

```rust

pub enum WeakRenderObject {
    RenderBox(WeakRenderBox),
    RenderSliver(WeakRenderSliver),
    RenderView(WeakRenderView),
}


impl WeakRenderObject {
    delegate::delegate!(to match self {
        WeakRenderObject::RenderBox(o) => o,
        WeakRenderObject::RenderSliver(o) => o,
        WeakRenderObject::RenderView(o) => o,
    } {
        pub fn is_alive(&self) -> bool;
    });
}
```

It is expanded as

```rust

impl WeakRenderObject {
    #[inline(always)]
    pub fn is_alive(&self) -> bool {
        match self {
            WeakRenderObject::RenderBox(o) => o.is_alive(),
            WeakRenderObject::RenderSliver(o) => o.is_alive(),
            WeakRenderObject::RenderView(o) => o.is_alive(),
        }
    }
}
```

**Comment by Kobzol** [comment]
Hi! I'm not sure what exactly do you mean. The code that you have posted seems to work, and expands to this:
```rust
#[inline(always)]
pub fn leading_ones(&self) -> u32 {
    match self {
        WeakRenderObject::RenderBox(o) => o,
        WeakRenderObject::RenderSliver(o) => o,
        WeakRenderObject::RenderView(o) => o,
    }.leading_ones()
}
```
Is there something problematic with this?

**Comment by gfreezy** [comment]
```
pub enum WeakRenderObject {
    RenderBox(WeakRenderBox),
    RenderSliver(WeakRenderSliver),
    RenderView(WeakRenderView),
}

struct WeakRenderBox;
struct WeakRenderSliver;
struct WeakRenderView;
```

`WeakRenderBox` `WeakRenderSliver` `WeakRenderView` are of different types. The `match` expr returns different types, thus compiles error.

**Comment by Kobzol** [comment]
I see. I'm not sure what could be the ideal way to enable this use-case though :thinking: In theory, there could be some placeholder value in the target expression and the delegation macro would replace that placeholder, like this:
```rust
impl WeakRenderObject {
    delegate::delegate!(to match self {
        WeakRenderObject::RenderBox(o) => o.$delegate,
        WeakRenderObject::RenderSliver(o) => o.$delegate,
        WeakRenderObject::RenderView(o) => o.$delegate,
    } {
        pub fn is_alive(&self) -> bool;
    });
}
```
but I'm not sure if I can parse the code well enough so that I can find all of the placeholders. I'll try.

**Comment by Kobzol** [comment]
Published `v0.8.0` with this feature. Thank you very much!
